certz/certz.proto

```diff
@@ -636,8 +641,84 @@ message CertificateSigningRequest {
 // the specified parameters.
 message GenerateCSRRequest {
   CSRParams params = 1;
+
+  // Indicates measurements that are prerequisites for certificate
+  // provisioning.
+  repeated ReferenceIntegritySpec integrity = 2;
 }
+
 // A message sent by the target as response to the GenerateCSRRequest.
 message GenerateCSRResponse {
   CertificateSigningRequest certificate_signing_request = 1;
+
+  // Reference integrity measurements as specified in the request.
+  repeated ReferenceIntegrityData integrity = 2;
+}
+
+message GetIntegrityManifestRequest {}
+
+message GetIntegrityManifestResponse {
+  bytes manifest = 1;
+}
+
+// See the TCG Reference Integrity Manifest (RIM) Information Model.
+message ReferenceIntegritySpec {
+  // Identifies the Root of Trust component.
+  string rot_id = 1;
+
+  oneof type {
```

Comment:
what would be the other types we would be considering here?
is this really a oneof?

Reply:
I think the idea is that there could be alternatives to MBM, so the oneof leaves that possibility open.